lib/name.mli

```diff
@@ -27,11 +27,20 @@ open Cstruct
 type label
 
 (** Domain name, as a list of strings. *)
-type domain_name = string list
+type domain_name
 
 (** Domain name map *)
 module Map: Map.S with type key = domain_name
 
+(** Create null domain name *)
+val empty: domain_name
+
+(** Render a {! domain_name} to a {! string} list. *)
+val domain_name_to_string_list : domain_name -> string list
```

Comment:
Prefer "to_string_list" and comment like "[to_string_list name] is the list of labels in the name [name]"